compiler/dyno/include/chpl/queries/Flags.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_QUERIES_FLAGS_H
+#define CHPL_QUERIES_FLAGS_H
+
+#include "chpl/queries/UniqueString.h"
+#include "chpl/queries/update-functions.h"
+#include "chpl/util/hash.h"
+
+namespace chpl {
+
+
+/**
+  This class represents a collection of compiler flags. Not every flag
+  present in an invocation of 'chpl --help' will be present here. This
+  class is intended to be a convenient place to store various different
+  boolean flags the compiler may need to check.
+*/
+class Flags final {
```

Comment:
IMO this one would be better named CompilationFlags or something like that. I feel "Flags" is too general (I was expecting it to be a bit vector, with that name).

Reply:
I agree, I'm fine with renaming it CompilationFlags as well.